sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -656,6 +660,64 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
     }
   }
 
+  /**
+   * Convert cudf unsigned integer to wider signed integer that parquet expects
+   * After spark 3.2.0, parquet read uint8 as int16, uint16 as int32, uint32 as int64
+   *
+   * @param inputTable the input table
+   * @return converted table if the input table has unsigned integer
+   */
+  private def convertToUnsigned(inputTable: Table) = {
```

Comment:
This is not needed if you integrate the unigned casting into the original columns iterating logic.

Reply:
done